presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -464,19 +468,21 @@ public void testNotMergeDifferentOrdering()
                                         .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
                                 window(windowMatcherBuilder -> windowMatcherBuilder
                                                 .specification(specificationA)
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(DISCOUNT_ALIAS))),
+                                                .addFunction(functionCall("sum", UNSPECIFIED_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
```

Comment:
I couldn't find another way to make it work unless and until the window matcher code is rewritten. Since the `window` in `WindowNode.Function.FunctionCall.window` is now abstract (I haven't yet understood why that field is needed, btw), the `WindowNode.Function.FunctionCall` doesn't have the window frame information (earlier, this field was structurally equivalent to WindowSpecification and hence carried the window frame information). Also, the place where this `WindowNode.Function.FunctionCall` is constructed, there is no window resolution information to resolve the window name into it's specification. I think we need to take another look at window matcher stuff later to fix this.